### DISCUSSION TITLE
Requests timing out in the background on iOS

### DISCUSSION THREAD

**Discussion by svenlombaert** [vercel/swr#1701] [Q&A]
Hi!

I'm experiencing this annoying issue on react-native and I'm not sure how to tackle it. There's a mode in our app where we set a `refreshInterval` on a couple of requests. This all works fine, until the user puts the app in the background. When coming back to the foreground a timeout error is triggered, thus triggering the `onError` callback of `swr`. Anything we can do to prevent this?

This illustrates what happens. There's quite the time gap between the last two api requests, and this is when the user comes back from the background.

<img width="1415" alt="Screenshot 2021-12-10 at 11 57 33" src="https://user-images.githubusercontent.com/2846986/145563359-7af819ed-96ee-4fa3-acea-0514d2285eae.png">

**Comment by shuding** [comment]
Hmm that's weird, I've seen a similar issue but it's on the web and it was caused by an early alpha version of React 18. If it's hard to debug I suggest disabling focus revalidation.
